src/Crypto/Gpgme/Key.hs

```diff
@@ -26,13 +28,26 @@ import System.IO.Unsafe
 import Crypto.Gpgme.Types
 import Crypto.Gpgme.Internal
 
--- | Returns a list of known 'Key's from the @context@.
+-- | Returns a list of all known 'Key's from the @context@.
 listKeys :: Ctx            -- ^ context to operate in
          -> IncludeSecret  -- ^ whether to include the secrets
          -> IO [Key]
-listKeys (Ctx {_ctx=ctxPtr}) secret = do
+listKeys ctx secret = listKeys' ctx secret nullPtr
+
+-- | Returns a list of known 'Key's from the @context@ that match a given pattern.
+searchKeys :: Ctx            -- ^ context to operate in
+           -> IncludeSecret  -- ^ whether to include the secrets
+           -> String         -- ^ The pattern to look for
```

Comment:
would you mind going into detail what exactly a pattern could be? When I think of patterns, I think of something with an asterisk that matches many. However when I read the docs on this function in the gpgme manual, it said that

> Patterns should be used to restrict the search to a certain common name or user, not to list many specific keys at once by listing their fingerprints or key IDs. 

So would you mind expanding on the docs a bit such that a user would know what type of string could be passed here?

Reply:
By "pattern" I mean pattern as described in the [GPGME manual](https://www.gnupg.org/documentation/manuals/gpgme/Listing-Keys.html):
> If pattern or *pattern is NULL, all available keys are returned. Otherwise, pattern is a NULL terminated array of strings that are used to limit the list to all keys matching at least one of the patterns verbatim. Note that the total length of all patterns is restricted to an engine-specific maximum (the exact limit also depends on the number of patterns and amount of quoting required, but a couple of hundred characters are usually accepted). Patterns should be used to restrict the search to a certain common name or user, not to list many specific keys at once by listing their fingerprints or key IDs.